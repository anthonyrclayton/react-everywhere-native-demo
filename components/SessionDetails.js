import React, { Component } from 'react';
import { ScrollView, Image, Modal, Text, TouchableOpacity, Platform } from 'react-native'
import styles from './styles/SessionStyles'

export default class SessionDetails extends Component {
  render() {
    const gravatarUrl = this.props.Speakers[0].GravatarUrl
    const margin = Platform.OS === "ios" ? 65 : 30

    return(
      <ScrollView style={{padding: 30, marginTop: margin}}>
        <Text style={styles.sessionTitle}>{this.props.Title}</Text>
        <Image source={{uri: `https:${gravatarUrl}`}} style={{width: 90, height: 90, marginTop: 30}}/>

        <Text style={{paddingTop: 20, paddingBottom: 20}}>{this.props.Abstract}</Text>
      </ScrollView>
    )
  }
}
