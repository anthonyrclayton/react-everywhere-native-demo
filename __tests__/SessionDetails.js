import 'react-native';
import React from 'react';
import SessionDetails from '../components/SessionDetails'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const session = {"Id":5445,"SessionTime":"0001-01-01T00:00:00","SessionStartTime":"2017-01-10T08:00:00","SessionEndTime":"2017-01-10T12:00:00","Room":null,"Rooms":["Indigo Bay"],"Title":"7 Languages in 7 Hours","Abstract":"Hands-on experience.","SessionType":"Pre-Compiler","Tags":["Ruby\/Rails","Testing","Functional Programming","Other"],"Category":"Programming Principles","Speakers":[{"Id":"df1a9b69-1b7a-4224-bf28-2796a6cbed61","FirstName":"Amber","LastName":"Conville","GravatarUrl":"\/\/www.gravatar.com\/avatar\/7c2a83b93fe0e5f2ed4c95c59367259c"}]}

it('renders correctly', () => {
  const tree = renderer.create(
    <SessionDetails {...session} />
  );
  expect(tree).toMatchSnapshot();
});
