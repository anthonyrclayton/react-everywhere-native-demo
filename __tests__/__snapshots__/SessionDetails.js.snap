exports[`test renders correctly 1`] = `
<ScrollView
  style={
    Object {
      "marginTop": 65,
      "padding": 30,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "bold",
      }
    }>
    7 Languages in 7 Hours
  </Text>
  <Image
    source={
      Object {
        "uri": "https://www.gravatar.com/avatar/7c2a83b93fe0e5f2ed4c95c59367259c",
      }
    }
    style={
      Object {
        "height": 90,
        "marginTop": 30,
        "width": 90,
      }
    } />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }>
    Hands-on experience in seven different languages, ranging from object-oriented to functional, from the cozy and familiar to the eye-squintingly terse and foreign. You\'ll work through a familiar kata in each of them, showing you how to approach an entirely new language. You\'ll learn about the strengths and weaknesses of each language out in the real world. You\'ll also see how each language\'s quirks can teach us ways to improve the code we write every day.
    
    You may not come out of this pre-compiler an expert in all of these languages, but you\'ll have learned a lot about how to get started with a new one. You may even discover a new passion! At the very least, the next time a new language comes along, you\'ll have the tools you need to tackle it, and enough knowledge to help you push past the \"what is this syntax even doing\" barrier.
    
    Languages: Ruby, Clojure, Haskell, Rust, Scala, Elixir, Go
  </Text>
</ScrollView>
`;
